### PR TITLE
Work with full HIT parser path

### LINKED ISSUE #13793: TestHarness printout should handle new nesting layers in test spec
## Reason

The MOOSEDocs system recently augmented the available syntax for Test Specs making another layer possible

e.g.
```
[Tests]
  [some_requirement]
    requirement = 'The system shall do this thing:'
  
    [1D]
       ...
       detail = 'in 1D, and'
    []
    [2D]
       ...
       detail = 'in 1D.'
    []
  []
[]
```

It would be handy if the way we printout the test included these levels.

## Design

Make that TestHarness merge all of the levels of the testing hierarchy in the printout. Right now the middle level (in the example above "some_requirement") does not appear in the printout.

## Impact

Minor: Usability

### PR DESCRIPTION
**Pull request by milljm**
Make the full path of the hit parser node tree available to
the tester object. This allows us to understand the path
hierarchy.

Closes #13793

### CODE REVIEW

**Comment by milljm** [comment]
This seems to break Pronghorn and Mammoth. I need to investigate.

**Comment by milljm** [comment]
A fix is in place to add the missing Test block to several tests in pronghorn and mammoth.

File: python/TestHarness/schedulers/Job.py [hunk 1]
```diff
@@ -350,8 +350,8 @@ def getJointStatus(self):
         Return the most accurate message possible, based on the statuses of both the Tester and Job.
         Most of the time, we want to return a Tester status. As this class will have the most detail.
         """
-        # Tester has no status, use a job status instead
-        if self.isTimeout() or self.isNoStatus():
+        # Job has failed, or tester has no status
+        if self.isError() or self.isNoStatus():
```
Comment (permcody) [inline python/TestHarness/schedulers/Job.py]: This change seems unrelated. This is intentional correct?

Comment (milljm) [reply python/TestHarness/schedulers/Job.py]: That was concerning the comment I made in commit 6bb2201:

Fix an interesting issue discovered during this process.
- A skipped test, which has failed due to dependencies was not
being detected properly.

**Review by permcody — APPROVED** [review]
